ui/src/app/kvm/components/KVMConfigurationCard/KVMConfigurationCard.test.tsx

```diff
@@ -1,126 +1,202 @@
+      target: { value: "3" },
+    });
+    fireEvent.change(screen.getByRole("combobox", { name: "Resource pool" }), {
+      target: { value: "2" },
+    });
+    fireEvent.change(screen.getByLabelText("Password (optional)"), {
+      target: { value: "password" },
+    });
+    fireEvent.change(screen.getByRole("slider", { name: "CPU overcommit" }), {
+      target: { value: "5" },
+    });
+    fireEvent.change(
+      screen.getByRole("slider", { name: "Memory overcommit" }),
+      { target: { value: "7" } }
     );
-    expect(
-      store.getActions().find((action) => action.type === "pod/update")
-    ).toStrictEqual({
-      type: "pod/update",
-      meta: {
-        method: "update",
-        model: "pod",
-      },
-      payload: {
-        params: {
-          cpu_over_commit_ratio: 2,
-          id: 1,
-          memory_over_commit_ratio: 2,
-          pool: 1,
-          power_address: "192.168.1.1",
-          power_pass: undefined,
-          tags: "tag1,tag2",
-          zone: 2,
-        },
-      },
+  });
+  await waitFor(() => {
+    fireEvent.click(screen.getByRole("button", { name: "Save changes" }));
+  });
+
+  const expectedAction = podActions.update({
+    cpu_over_commit_ratio: 5,
+    id: pod.id,
+    memory_over_commit_ratio: 7,
+    pool: 2,
+    power_address: pod.power_parameters.power_address,
+    power_pass: "password",
+    tags: "tag1,tag2",
+    type: PodType.VIRSH,
+    zone: 3,
+  });
+  expect(
+    store.getActions().find((action) => action.type === expectedAction.type)
+  ).toStrictEqual(expectedAction);
+});
+
+it("enables the submit button if form is unchanged after pod successfully updated", async () => {
+  const pod = podFactory({
+    cpu_over_commit_ratio: 1,
+    id: 1,
+    type: PodType.LXD,
+  });
+  const store = mockStore(state);
+  const { rerender } = render(
+    <Provider store={store}>
+      <MemoryRouter
+        initialEntries={[{ pathname: "/kvm/1/edit", key: "testKey" }]}
+      >
+        <KVMConfigurationCard pod={pod} />
+      </MemoryRouter>
+    </Provider>
+  );
+
+  // Submit should be disabled by default.
+  expect(screen.getByRole("button", { name: "Save changes" })).toBeDisabled();
+
+  // Change value to something other than the initial.
+  await waitFor(() => {
+    fireEvent.change(screen.getByRole("slider", { name: "CPU overcommit" }), {
+      target: { value: (pod.cpu_over_commit_ratio + 1).toString() },
     });
   });
 
-  it("can handle updating a virsh KVM", () => {
-    const pod = podFactory({ id: 1, type: PodType.VIRSH });
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/kvm/1/edit", key: "testKey" }]}
-        >
-          <KVMConfigurationCard pod={pod} />
-        </MemoryRouter>
-      </Provider>
-    );
+  // Submit should be enabled.
+  expect(
+    screen.getByRole("button", { name: "Save changes" })
+  ).not.toBeDisabled();
```

Comment:
Similar to the comment above, `waitFor` wrapper should probably be used around this expect instead of around fireEvent.